### ISSUE TITLE
Create "native" Engine API for low-level use cases

### ISSUE THREAD

**Issue by mr-smidge** [xsco/libdjinterop#66]
# Overview

The public API in libdjinterop is primarily targeted as being a format-neutral design: it is not meant to encompass concepts that are specific to just a single format, but provide an API that could broadly work across any DJ library format.

As a result, there will be times when something specific to a particular format is needed by a user, but cannot be accessed through the format-neutral API.  To do so, it would be beneficial to also offer a "native" API offering access to the lower-level details of a specific format, in terms defined by that format.

For example, an Engine 1.x API might offer types based broadly on the database tables in the music and performance SQLite databases.  A user would be able to access (and modify) data not normally exposed.  To some extent, the `storage` class idea encapsulates some of this already, although that class is not designed to be exposed to end users.

Such an API could be made available through includes under the `djinterop::engine` namespace.

In implementation terms, the `*_impl` classes could delegate to the native API, making the former purely about format-neutrality conversion and not the specifics of, say, persistence.

# Acceptance Criteria

* The library _must_ offer a "native" Engine API, in addition to the format-neutral API.

**Comment by mr-smidge** [comment]
PR #80 effectively completes this for Engine v2.  With Engine v3 needed soon, and with Engine v1 being considered defunct, it seems low priority to consider this issue to only be closed once a low-level API for Engine v1 is in place.

As such, considering this complete.
